--- a/externals/coda-oss/modules/c++/sys/include/sys/Conf.h
+++ b/externals/coda-oss/modules/c++/sys/include/sys/Conf.h
@@ -1 +1 @@
-#ifndef _WIN32
-#include <features.h>
-#endif
+#ifndef _WIN32
+#  if defined(__has_include)
+#    if __has_include(<features.h>)
+#      include <features.h>
+#    endif
+#  elif defined(__linux__)
+#    include <features.h>
+#  endif
+#endif
@@ -1 +1 @@
 #undef CODA_OSS_POSIX_SOURCE
 #if defined(_POSIX_C_SOURCE) && (_POSIX_C_SOURCE >= 1)
 #define CODA_OSS_POSIX_SOURCE _POSIX_C_SOURCE
+#elif defined(__APPLE__)
+#define CODA_OSS_POSIX_SOURCE 200809L
 #endif

 #undef CODA_OSS_POSIX2001_SOURCE
 #if defined(CODA_OSS_POSIX_SOURCE) && (_POSIX_C_SOURCE >= 200112L)
 #define CODA_OSS_POSIX2001_SOURCE _POSIX_C_SOURCE
+#elif defined(__APPLE__)
+#define CODA_OSS_POSIX2001_SOURCE 200809L
 #endif

 #undef CODA_OSS_POSIX2008_SOURCE
 #if defined(CODA_OSS_POSIX2001_SOURCE) && (_POSIX_C_SOURCE >= 200809L)
 #define CODA_OSS_POSIX2008_SOURCE _POSIX_C_SOURCE
+#elif defined(__APPLE__)
+#define CODA_OSS_POSIX2008_SOURCE 200809L
 #endif