--- a/externals/coda-oss/modules/c++/coda_oss/include/coda_oss/bit.h
+++ b/externals/coda-oss/modules/c++/coda_oss/include/coda_oss/bit.h
@@ -23,9 +23,10 @@
 #define CODA_OSS_coda_oss_bit_h_INCLUDED_
 #include <stdlib.h>
 #include <stdint.h>
-#ifdef __GNUC__
-#include <byteswap.h>  // "These functions are GNU extensions."
-#endif
+#if defined(__has_include) && __has_include(<byteswap.h>)
+#  include <byteswap.h>
+#  define CODA_OSS_HAS_BSWAP_BUILTINS 1
+#endif
 #include <type_traits>
 #include "coda_oss/namespace_.h"
 namespace coda_oss
@@ -66,19 +67,32 @@
     {
         return _byteswap_uint64(val);
     }
-    #elif defined(__GNUC__)
+    #elif defined(CODA_OSS_HAS_BSWAP_BUILTINS)
     inline uint16_t byteswap(uint16_t val) noexcept
     {
         return bswap_16(val);
     }
     inline uint32_t byteswap(uint32_t val) noexcept
     {
         return bswap_32(val);
     }
     inline uint64_t byteswap(uint64_t val) noexcept
     {
         return bswap_64(val);
     }
+    #elif defined(__GNUC__) || defined(__clang__)
+    inline uint16_t byteswap(uint16_t val) noexcept
+    {
+        return __builtin_bswap16(val);
+    }
+    inline uint32_t byteswap(uint32_t val) noexcept
+    {
+        return __builtin_bswap32(val);
+    }
+    inline uint64_t byteswap(uint64_t val) noexcept
+    {
+        return __builtin_bswap64(val);
+    }
     #else
     #error "No platform-specific byteswap()" // TODO: do something else?
     #endif